src/metabase/lib/drill_thru/underlying_records.cljc

```diff
@@ -186,20 +187,25 @@
         ;; The column-ref should be an aggregation ref - look up the full aggregation.
         aggregation (when-let [agg-uuid (last column-ref)]
                       (m/find-first #(= (lib.options/uuid %) agg-uuid)
-                                    (lib.aggregation/aggregations query -1)))]
-    ;; Apply the filters derived from the aggregation.
-    (reduce #(lib.filter/filter %1 -1 %2)
-            filtered
-            ;; If we found an aggregation, check if it implies further filtering.
-            ;; Simple aggregations like :sum don't add more filters; metrics or fancy aggregations like :sum-where do.
-            (when aggregation
-              (case (first aggregation)
-                ;; Fancy aggregations that filter the input - the filter is the last part of the aggregation.
-                (:sum-where :count-where :share)
-                [(last aggregation)]
-
-                ;; Default: no filters to add.
-                nil)))))
+                                    (lib.aggregation/aggregations query -1)))
+        ;; Apply the filters derived from the aggregation.
+        agg-filtered (reduce #(lib.filter/filter %1 -1 %2)
+                             filtered
+                             ;; If we found an aggregation, check if it implies further filtering.
+                             ;; Simple aggregations like :sum don't add more filters; metrics or fancy aggregations like :sum-where do.
+                             (when aggregation
+                               (case (first aggregation)
+                                 ;; Fancy aggregations that filter the input - the filter is the last part of the aggregation.
+                                 (:sum-where :count-where :share)
+                                 [(last aggregation)]
+
+                                 ;; Default: no filters to add.
+                                 nil)))
+        ;; make all joins include all fields
+        new-joins (map #(lib.join/with-join-fields % :all)
+                       (lib.join/joins agg-filtered))]
```

Comment:
As is, if a join has explicit fields already selected, this will overwrite that and display all fields.  I could include a check to only set fields to all if there are no fields currently selected, but I'm not sure if that would come up or be correct.

Reply:
I think that's fine. In the context of this `underlying-records` drill, we just removed the aggregations and breakouts, and having summaries means that the `:fields` of the source and joins had already been overwritten.